packages/vue-components/src/TipBox.vue

```diff
@@ -251,61 +257,29 @@ export default {
       return {};
     },
     getBootstrapAlertStyle() {
-      switch (this.type) {
-      case 'warning':
-        return 'alert-warning';
-      case 'info':
-        return 'alert-info';
-      case 'definition':
-        return 'alert-primary';
-      case 'success':
-      case 'tip':
-        return 'alert-success';
-      case 'important':
-      case 'wrong':
-        return 'alert-danger';
-      default:
-        return 'alert-default';
+      const { baseStyle, colorStyle } = classifyBootstrapStyle(this.type, this.theme);
+      if (colorStyle) {
+        return `alert-${colorStyle}`;
       }
+      return `alert-${baseStyle}`;
     },
     getBootstrapTextStyle() {
-      switch (this.type) {
-      case 'warning':
-        return 'text-warning';
-      case 'info':
-        return 'text-info';
-      case 'definition':
-        return 'text-primary';
-      case 'success':
-      case 'tip':
-        return 'text-success';
-      case 'important':
-      case 'wrong':
-        return 'text-danger';
-      default:
-        return '';
+      const { baseStyle, colorStyle } = classifyBootstrapStyle(this.type, this.theme);
+      if (colorStyle) {
+        return `text-${colorStyle}`;
       }
+      return `text-${baseStyle}`;
     },
     getBootstrapBorderStyle() {
-      switch (this.type) {
-      case 'warning':
-        return 'border-warning';
-      case 'info':
-        return 'border-info';
-      case 'definition':
-        return 'border-primary';
-      case 'success':
-      case 'tip':
-        return 'border-success';
-      case 'important':
-      case 'wrong':
-        return 'border-danger';
-      default:
-        return '';
+      const { baseStyle, colorStyle } = classifyBootstrapStyle(this.type, this.theme);
+      if (colorStyle) {
+        return `border-${colorStyle}`;
       }
+      return `border-${baseStyle}`;
     },
     getFontAwesomeIconStyle() {
-      switch (this.type) {
+      const { iconStyle } = classifyBootstrapStyle(this.type, this.theme);
+      switch (iconStyle) {
```

Comment:
What do you think about extracting this to `classifyBootstrapStyle` as well?

Reply:
Ok this does make more sense, will make this change